### PR TITLE
[clickhouse-cpp] Add new port

### PR DESCRIPTION
**Pull request by zi-m**
https://github.com/artpaul/clickhouse-cpp

### CODE REVIEW

File: ports/clickhouse-cpp/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,28 @@
+include(vcpkg_common_functions)
+
+vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
```
Comment (dan-shaw) [inline ports/clickhouse-cpp/portfile.cmake]: Can this library be built dynamically?

File: ports/clickhouse-cpp/00001-fix-build.patch [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+index 7a17e6d..cb29d1d 100644
+--- a/clickhouse/base/socket.cpp
++++ b/clickhouse/base/socket.cpp
+@@ -34,6 +34,23 @@ public:
+     }
+ };
+ 
++// from https://github.com/catboost/catboost/blob/master/util/network/socket.cpp
++#if defined(_win_)
++static inline void SetNonBlockSocket(SOCKET fd, int value) {
```
Comment (dan-shaw) [inline ports/clickhouse-cpp/00001-fix-build.patch]: Is this change needed? Usually, we don't want to add a new function in the source.

**Review by dan-shaw — COMMENTED** [review]
1 more comment, otherwise LGTM

**Comment by dan-shaw** [comment]
Is the `cityhash` usage file needed? vcpkg can generate this.

Otherwise, looks good to merge.

**Comment by zi-m** [comment]
> Is the `cityhash` usage file needed? vcpkg can generate this.
> 
> Otherwise, looks good to merge.

Ups! Removed.

**Comment by dan-shaw** [comment]
Thanks for the PR!
